lib/Sema/TypeCheckPattern.cpp

```diff
@@ -738,9 +738,14 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
                                   TypeChecker &TC) {
   if (auto ty = decl->getTypeLoc().getType())
     return ty->is<ErrorType>();
-  
+
+  // If the element is a variadic parameter, resolve the parameter type as if
+  // it were in non-parameter position, since we want functions to be
+  // @escaping in this case.
```

Comment:
This seems a bit too subtle and easy to break. I think it would be better to check `TR_ImmediateFunctionInput` instead of `TR_FunctionInput`, no?